pyptex/__init__.py

```diff
@@ -229,298 +241,305 @@ def genname(self,pattern="fig{gencount}.eps"):
-          uses something like
+          uses something like.
           `r"{latex} {pyptexfilename} && (test ! -f {bibfilename} || bibtex {auxfilename})"`
-          The relevant substitutions are performed by `string.format` from `pyp.__dict__`
+          The relevant substitutions are performed by `string.format` from `pyp.__dict__`.
         * `pyp.disable_cache = False`, set this to `True` if you want to disable the `a.pickle`
           cache. You shouldn't need to do this but if your Python code is nondeterministic
           or if tracking dependencies is too hard, disabling all caching will ensure
           that `a.pyptex` is correctly compiled into `a.pdf` and that a stale cache is
           never used.
-        * `pyp.deps` is a dictionary of dependencies and timestamps
-        * `pyp.lc` counts lines while parsing
-        * `pyp.argv` stores the ``command-line arguments'' for template generation
-        * `pyp.exitcode` is the exit code of the `pyp.latexcommand`
-        * `pyp.gencount` is the counter for generated files (see `pyp.gen()`)
-        * `pyp.fragments` is the list of Python fragments extracted from a.tex
+        * `pyp.deps` is a dictionary of dependencies and timestamps.
+        * `pyp.lc` counts lines while parsing.
+        * `pyp.argv` stores the ``command-line arguments'' for template generation.
+        * `pyp.exitcode` is the exit code of the `pyp.latexcommand`.
+        * `pyp.gencount` is the counter for generated files (see `pyp.gen()`).
+        * `pyp.fragments` is the list of Python fragments extracted from a.tex.
         * `pyp.outputs` is the matching outputs.
-        * `pyp.compiled` is the string that is written to `a.pyptex`
+        * `pyp.compiled` is the string that is written to `a.pyptex`.
         """
-        print(texfilename+": pyptex compilation begins")
-        self.__globals__ = {"__builtins__":__builtins__,"pyp":weakref.proxy(self)}
+        print(f'{texfilename}: pyptex compilation begins')
+        self.__globals__ = {'__builtins__': __builtins__, 'pyp': weakref.proxy(self)}
         self.filename = os.path.splitext(texfilename)[0]
         self.texfilename = texfilename
-        self.cachefilename = self.filename+'.pickle'
-        self.bibfilename = self.filename+'.bib'
-        self.pyptexfilename = self.filename+'.pyptex'
-        self.auxfilename = self.filename+'.aux'
-        self.latex = "pdflatex --file-line-error --synctex=1"
+        self.cachefilename = f'{self.filename}.pickle'
+        self.bibfilename = f'{self.filename}.bib'
+        self.pyptexfilename = f'{self.filename}.pyptex'
+        self.auxfilename = f'{self.filename}.aux'
+        self.latex = 'pdflatex --file-line-error --synctex=1'
         self.latexcommand = latexcommand
         self.disable_cache = False
         self.deps = {}
         self.lc = 0
-        self.argv = argv
+        self.argv = [] if argv is None else []
```

Comment:
Why did this happen here, why not just make the default value []?

Is there a point where the default value [] is explicitly overridden with None but you would want self.argv to be [] instead of the explicitly provided None?

Perhaps self.argv has a confusing name but I don't think I intended to override people's wishes if they don't want to use a list. None would be a perfectly fine value for argv. I guess I didn't unit test whether it works though, maybe I did assume it's not "None".
```suggestion
        self.argv = argv
```

Reply:
Referenced up [here](https://github.com/sloisel/pyptex/pull/2#discussion_r389392377).